pkg/storage/unified/apistore/prepare.go

```diff
@@ -47,7 +47,7 @@ func (s *Storage) prepareObjectForStorage(ctx context.Context, newObject runtime
 		return nil, err
 	}
 	if obj.GetName() == "" {
-		return nil, storage.ErrResourceVersionSetOnCreate
+		return nil, storage.NewInvalidObjError("", "missing name")
```

Comment:
Unrelated to this PR, but when adding a unit test, this shouldn't be returning that the RV was set